batch/batch/front_end/front_end.py

```diff
@@ -1301,8 +1325,9 @@ async def _query_billing_projects(db, user=None, billing_project=None):
         where_condition = ''
 
     sql = f'''
-SELECT billing_projects.name as billing_project, users
-FROM (
+SELECT billing_projects.name as billing_project,
+billing_projects.`status` as `status`,
+users FROM (
   SELECT billing_project, JSON_ARRAYAGG(`user`) as users
   FROM billing_project_users
   GROUP BY billing_project
```

Comment:
Hmm, (not your problem, but) I think this query can also be written without a sub-query:
```sql
SELECT billing_projects.name as billing_project,
       billing_projects.`status` as `status`,
       IF (COUNT(`user`) = 0, JSON_ARRAY(), JSON_ARRAYAGG(`user`)) as users
FROM billing_projects
LEFT JOIN billing_project_users
ON billing_projects.name = billing_project_users.billing_project
GROUP BY billing_projects.name, `status`
{where_condition}
```
```
mysql> explain SELECT billing_projects.name as billing_project, users
    -> FROM (
    ->   SELECT billing_project, JSON_ARRAYAGG(`user`) as users
    ->   FROM billing_project_users
    ->   GROUP BY billing_project
    -> ) AS t
    -> RIGHT JOIN billing_projects
    ->   ON t.billing_project = billing_projects.name
    -> ;
+----+-------------+-----------------------+------------+-------+---------------+------------------------+---------+------+------+----------+----------------------------------------------------+
| id | select_type | table                 | partitions | type  | possible_keys | key                    | key_len | ref  | rows | filtered | Extra                                              |
+----+-------------+-----------------------+------------+-------+---------------+------------------------+---------+------+------+----------+----------------------------------------------------+
|  1 | PRIMARY     | billing_projects      | NULL       | index | NULL          | billing_project_status | 1       | NULL |   25 |   100.00 | Using index                                        |
|  1 | PRIMARY     | <derived2>            | NULL       | ALL   | NULL          | NULL                   | NULL    | NULL |    6 |   100.00 | Using where; Using join buffer (Block Nested Loop) |
|  2 | DERIVED     | billing_project_users | NULL       | index | PRIMARY       | PRIMARY                | 604     | NULL |    6 |   100.00 | Using index; Using filesort                        |
+----+-------------+-----------------------+------------+-------+---------------+------------------------+---------+------+------+----------+----------------------------------------------------+
3 rows in set, 1 warning (0.00 sec)

mysql> explain SELECT billing_projects.name as billing_project,
    ->        billing_projects.`status` as `status`,
    ->        IF (COUNT(`user`) = 0, JSON_ARRAY(), JSON_ARRAYAGG(`user`)) as users
    -> FROM billing_projects
    -> LEFT JOIN billing_project_users
    -> ON billing_projects.name = billing_project_users.billing_project
    -> GROUP BY billing_projects.name, `status`
    -> ;
+----+-------------+-----------------------+------------+-------+------------------------+------------------------+---------+----------------------------+------+----------+-----------------------------+
| id | select_type | table                 | partitions | type  | possible_keys          | key                    | key_len | ref                        | rows | filtered | Extra                       |
+----+-------------+-----------------------+------------+-------+------------------------+------------------------+---------+----------------------------+------+----------+-----------------------------+
|  1 | SIMPLE      | billing_projects      | NULL       | index | billing_project_status | billing_project_status | 1       | NULL                       |   25 |   100.00 | Using index; Using filesort |
|  1 | SIMPLE      | billing_project_users | NULL       | ref   | PRIMARY                | PRIMARY                | 302     | wang.billing_projects.name |    1 |   100.00 | Using index                 |
+----+-------------+-----------------------+------------+-------+------------------------+------------------------+---------+----------------------------+------+----------+-----------------------------+
2 rows in set, 1 warning (0.01 sec)
```

The `IF` kind of sucks, but I think that's the only way to deal with the left joins.

Reply:
I'll PR this separately.